node/src/service.rs

```diff
@@ -257,10 +257,15 @@ async fn start_node_impl(
         // Putting a link in there and swapping out the requirements for your
         // own are probably a good idea. The requirements for a para-chain are
         // dictated by its relay-chain.
-        if !SUBSTRATE_REFERENCE_HARDWARE.check_hardware(&hwbench) && validator {
-            log::warn!(
-                "⚠️  The hardware does not meet the minimal requirements for role 'Authority'."
-            );
+        match SUBSTRATE_REFERENCE_HARDWARE.check_hardware(&hwbench) {
+            Err(err) if validator => {
+                log::warn!(
+                    "⚠️  The hardware does not meet the minimal requirements {} for role \
+                     'Authority'.",
+                    err
+                );
+            }
+            _ => {}
```

Comment:
```suggestion
        if let Err(err) = SUBSTRATE_REFERENCE_HARDWARE.check_hardware(&hwbench) {
            if validator {
                log::warn!(
                    "⚠️  The hardware does not meet the minimal requirements {} for role \
                     'Authority'.",
                    err
                );
            }
        }
```

Having an empty match arm feels like we are using the wrong tool. I think `if let(err)` suits better for this. Or if we do not need to extract the error, even a simple `if` with `&&` to combine two if statements.

Reply:
Hmm, i'm surprised we are on different sides about this one. Doesn't matter though, very minor thing. Not a blocker :)

Out of curiosity, if you can share the example from polkadot-sdk for the match (where one arm is empty and can be replaced with `if let Err()`, I'd be glad.